### PR TITLE
VIH-10460 Enabled V2 changes the V1 judge email and username

### PR DESCRIPTION
**Pull request by marcogagliardi**
[VIH-10460](https://tools.hmcts.net/jira/browse/VIH-10460)

added hearing Role Name to model

### CODE REVIEW

File: AdminWebsite/AdminWebsite/Mappers/ParticipantResponseMapper.cs [hunk 1]
```diff
@@ -39,6 +39,7 @@ public static List<ParticipantResponse> Map(this List<V2.ParticipantResponseV2>
                 Id = p.Id,
                 DisplayName = p.DisplayName,
                 HearingRoleCode = p.HearingRoleCode,
+                HearingRoleName = p.HearingRoleName,
```
Comment (oliver-scott) [inline AdminWebsite/AdminWebsite/Mappers/ParticipantResponseMapper.cs:42]: As with the bookings api PR, please can you add a test to verify that this property is mapped?

File: AdminWebsite/AdminWebsite.UnitTests/Mappers/ParticipantResponseMapperTest.cs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+namespace AdminWebsite.UnitTests.Mappers
+{
+    public class ParticipantResponseMapperTest
+    {
+        [Test]
+        public void Should_map_person_response_to_participant_response_V1()
+        {
+            var id = Guid.NewGuid();
+            List<BookingsApi.Contract.V1.Responses.ParticipantResponse> participants = new List<BookingsApi.Contract.V1.Responses.ParticipantResponse>();
```
Comment (oliver-scott) [inline AdminWebsite/AdminWebsite.UnitTests/Mappers/ParticipantResponseMapperTest.cs:18]: can use `var` here instead

File: AdminWebsite/AdminWebsite.UnitTests/Mappers/ParticipantResponseMapperTest.cs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+            BookingsApi.Contract.V1.Responses.ParticipantResponse participant = new BookingsApi.Contract.V1.Responses.ParticipantResponse()
```
Comment (oliver-scott) [inline AdminWebsite/AdminWebsite.UnitTests/Mappers/ParticipantResponseMapperTest.cs:19]: as per previous comment, use `var`

File: AdminWebsite/AdminWebsite.UnitTests/Mappers/ParticipantResponseMapperTest.cs [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+                ContactEmail = "judge@personal.com",
+                HearingRoleName = "Judge",
+                DisplayName = "Display Name",
+                Id = id
+            };
+            participants.Add(participant);
+
+            var participantsResponse = ParticipantResponseMapper.Map(participants);
+
+            foreach (var participantResponse in participantsResponse)
```
Comment (oliver-scott) [inline AdminWebsite/AdminWebsite.UnitTests/Mappers/ParticipantResponseMapperTest.cs:32]: could we assert on all of the mapped properties here?

Id
DisplayName
CaseRoleName
HearingRoleName
UserRoleName
Title
FirstName
MiddleNames
LastName
ContactEmail
TelephoneNumber
Username
Organisation
Representee
LinkedParticipants

File: AdminWebsite/AdminWebsite.UnitTests/Mappers/ParticipantResponseMapperTest.cs [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+                participantResponse.LastName.Should().Be(participant.LastName);
+                participantResponse.ContactEmail.Should().Be(participant.ContactEmail);
+                participantResponse.HearingRoleName.Should().Be(participant.HearingRoleName);
+                participantResponse.DisplayName.Should().Be(participant.DisplayName);
+                participantResponse.Id.Should().Be(participant.Id);
+            }
+        }
+        
+        [Test]
+        public void Should_map_person_response_to_participant_response_V2()
```
Comment (oliver-scott) [inline AdminWebsite/AdminWebsite.UnitTests/Mappers/ParticipantResponseMapperTest.cs:44]: see comments for the V1 test regarding var and the asserted properties

File: AdminWebsite/AdminWebsite.UnitTests/Controllers/HearingsController/EditMultiDayHearingTests.cs [hunk 6]
```diff
@@ -491,8 +491,7 @@ public async Task Should_forward_unhandled_error_from_bookings_api()
                     FirstName = x.FirstName,
                     LastName = x.LastName,
                     ContactEmail = x.ContactEmail,
-                    DisplayName = x.DisplayName,
-                    HearingRoleCode = x.HearingRoleName
+                    DisplayName = x.DisplayName
```
Comment (oliver-scott) [inline AdminWebsite/AdminWebsite.UnitTests/Controllers/HearingsController/EditMultiDayHearingTests.cs:494]: could you revert these changes please, you shouldn't need to change this test

**Review by oliver-scott — COMMENTED** [review]
thanks for adding the tests, left some comments

File: AdminWebsite/AdminWebsite.UnitTests/Mappers/ParticipantResponseMapperTest.cs [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+using BookingsApi.Contract.V1.Responses;
+using FluentAssertions;
+using NUnit.Framework;
+
+namespace AdminWebsite.UnitTests.Mappers
+{
+    public class ParticipantResponseMapperTest
+    {
+        [Test]
+        public void Should_map_person_response_to_participant_response_V1()
```
Comment (oliver-scott) [inline AdminWebsite/AdminWebsite.UnitTests/Mappers/ParticipantResponseMapperTest.cs:15]: suggested title - `Should_map_participant_response_V1`

File: AdminWebsite/AdminWebsite.UnitTests/Mappers/ParticipantResponseMapperTest.cs (hunk 5 above)
Comment (oliver-scott) [inline AdminWebsite/AdminWebsite.UnitTests/Mappers/ParticipantResponseMapperTest.cs:44]: suggested title - `Should_map_participant_response_V2`

File: AdminWebsite/AdminWebsite.UnitTests/Mappers/ParticipantResponseMapperTest.cs (hunk 4 above)
Comment (marcogagliardi) [reply AdminWebsite/AdminWebsite.UnitTests/Mappers/ParticipantResponseMapperTest.cs:32]: done

File: AdminWebsite/AdminWebsite.UnitTests/Mappers/ParticipantResponseMapperTest.cs (hunk 3 above)
Comment (marcogagliardi) [reply AdminWebsite/AdminWebsite.UnitTests/Mappers/ParticipantResponseMapperTest.cs:19]: done

File: AdminWebsite/AdminWebsite.UnitTests/Mappers/ParticipantResponseMapperTest.cs (hunk 2 above)
Comment (marcogagliardi) [reply AdminWebsite/AdminWebsite.UnitTests/Mappers/ParticipantResponseMapperTest.cs:18]: done

File: AdminWebsite/AdminWebsite.UnitTests/Mappers/ParticipantResponseMapperTest.cs (hunk 5 above)
Comment (marcogagliardi) [reply AdminWebsite/AdminWebsite.UnitTests/Mappers/ParticipantResponseMapperTest.cs:44]: done

File: AdminWebsite/AdminWebsite.UnitTests/Controllers/HearingsController/EditMultiDayHearingTests.cs (hunk 6 above)
Comment (marcogagliardi) [reply AdminWebsite/AdminWebsite.UnitTests/Controllers/HearingsController/EditMultiDayHearingTests.cs:494]: done

File: AdminWebsite/AdminWebsite.UnitTests/Mappers/ParticipantResponseMapperTest.cs (hunk 7 above)
Comment (marcogagliardi) [reply AdminWebsite/AdminWebsite.UnitTests/Mappers/ParticipantResponseMapperTest.cs]: done

**Review by komaldabhi1509 — APPROVED** [review]

**Review by oliver-scott — APPROVED** [review]
